Dockerfile.CI

```diff
@@ -1,43 +1,50 @@
-FROM node:14-alpine as builder-client
+# 1. BUILD FRONTEND source with webpack
+FROM node:14-alpine as builder-frontend
 ARG env=production
 ENV npm_config_cache=/tmp/.npm
-RUN mkdir -p /tmp/.npm /app \
-    && chown nobody:nobody /tmp/.npm /app
+RUN mkdir -p /tmp/.npm /app && chown nobody:nobody /tmp/.npm /app
 WORKDIR /app
 USER nobody
-
 ADD git_src/package.json git_src/package-lock.json ./
 RUN npm install
 ADD git_src/webpack.config.cjs git_src/babel.config.cjs ./
 ADD git_src/client ./client
 RUN npm run build-frontend
 
-
-FROM node:14-alpine as builder-app
+# 2. BUILD BACKEND source from .ts, uses dev-dependencies
+FROM node:14-alpine as builder-backend
 ARG env=production
 ENV npm_config_cache=/tmp/.npm
 RUN mkdir -p /app && chown nobody:nobody /app
-COPY --from=builder-client /tmp/.npm /tmp/.npm
 WORKDIR /app
-ADD git_src/package.json git_src/package-lock.json ./
+USER nobody
+ADD src ./src
+ADD git_src/package.json git_src/package-lock.json git_src/tsconfig.json ./
 RUN npm install
 RUN npm run build-backend
 
+# 3. BUILD FINAL IMAGE
 FROM node:14-alpine
 ARG env=production
 ENV NODE_ENV=${env}
 RUN mkdir -p /app && chown nobody:nobody /app
-USER nobody
 WORKDIR /app
-COPY git_src/ /app
+COPY --from=builder-frontend /app/dist /app/dist
+COPY --from=builder-backend /app/app /app/app
 
-# install  production dependencies only
-RUN npm install
+# 3.1 copy only required files
+COPY git_src/migrations /app/migrations
+COPY git_src/container-health.js /app/container-health.js
+COPY git_src/knexfile.js /app/knexfile.js
+COPY git_src/package.json /app/package.json
+COPY git_src/package-lock.json /app/package-lock.json
 
-COPY --from=builder-client /app/dist /app/dist
-COPY --from=builder-app /app/app /app/app
+# 3.2 install production dependencies only. Cleanup cache after that
+RUN mkdir -p /.npm && chown nobody:nobody /.npm
+RUN npm ci
+RUN rm -rf /.npm
```

Comment:
I recommend to concatenate to one layer, to make disk size more efficient
RUN mkdir -p /.npm && chown nobody:nobody /.npm \
 && npm ci \
 && rm -rf /.npm
 
I still don't understand why you change the owner of .npm folder, you delete that folder anyway.